task_processing/executors/task_executor.py

```diff
@@ -0,0 +1,43 @@
+# -*- coding: utf-8 -*-
+from __future__ import absolute_import
+from __future__ import unicode_literals
+
+import abc
+from collections import namedtuple
+
+import six
+
+TaskConfig = namedtuple(
```

Comment:
This seems very coupled. Perhaps we can register these with the framework or something (plugins)